ledger/shelley/genesis.go

```diff
@@ -42,7 +43,40 @@ type ShelleyGenesis struct {
 	ProtocolParameters ShelleyGenesisProtocolParams `json:"protocolParams"`
 	GenDelegs          map[string]map[string]string `json:"genDelegs"`
 	InitialFunds       map[string]uint64            `json:"initialFunds"`
-	Staking            any                          `json:"staking"`
+	Staking            GenesisStaking               `json:"staking"`
+}
+
+type GenesisStaking struct {
+	Pools map[string]GenesisPool `json:"pools"`
+	Stake map[string]string      `json:"stake"`
+}
+
+type GenesisPool struct {
+	Cost          int64          `json:"cost"`
+	Margin        float64        `json:"margin"`
+	Metadata      interface{}    `json:"metadata"`
+	Owners        []string       `json:"owners"`
+	Pledge        int64          `json:"pledge"`
+	PublicKey     string         `json:"publicKey"`
+	Relays        []GenesisRelay `json:"relays"`
+	RewardAccount GenesisReward  `json:"rewardAccount"`
+	Vrf           string         `json:"vrf"`
+}
+
+type GenesisRelay struct {
+	SingleHostName *SingleHostName `json:"single host name,omitempty"`
+}
+
+type SingleHostName struct {
+	DNSName string `json:"dnsName"`
+	Port    int    `json:"port"`
+}
+
+type GenesisReward struct {
+	Credential struct {
+		KeyHash string `json:"key hash"`
+	} `json:"credential"`
+	Network string `json:"network"`
 }
```

Comment:
There's a lot of overlap between these structures and `common.PoolRegistrationCertificate`, `common.PoolRelay`, and related types. We could probably merge them and avoid the need to convert/copy between them later.